pkgs/development/python-modules/ovirt-engine-sdk/default.nix

```diff
@@ -0,0 +1,37 @@
+{ 
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  pycurl,
+  six,
+  libxml2Python
+}:
+
+buildPythonPackage rec {
+  pname = "ovirt-engine-sdk-python";
+  version = "4.3.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0hz7rcz7l391n8ixcwvqw3iq5bx26fk9l5c6xxymv036zm3r9pnh";
+  };
+
+  doCheck = true;
```

Comment:
no tests are located in the pypi tarball, please use fetchFromGitHub and pull down the source so that tests are included.

Reply:
Github sources is a Java-based code generator which does not directly contain any tests other than those included in the build/generate process therefore it is safe to assume that the product (i.e. the PyPi lib) is already tested. This is also supported by the fact that what is on PyPi is actually a tagged release.

What's the point of more tests, especially when they're not directly provided by upstream in the first place?

There are many PyPi packages in nixpkgs with no tests.

Reply:
sure, but bumps in libxml2Python, pycurl or any of their dependencies may break this package.

The tests are to verify that the package is usable beyond a, "think so"

Reply:
Fair enough. Though I have no idea how to build the package from the upstream source. This may take some time.

Reply:
you should just need to do a:
```
sourceRoot = "source/sdk";
```
and everything else should be fine

Reply:
Doesn't look like it, `mvn package` still must be somehow ran before the actual python build because some of the code in `sdk/` must be first generated by that Maven build. At least that's my current level of understanding this.

Reply:
oh, really? hmm..